alkira/resource_alkira_service_f5_lb.go

```diff
@@ -0,0 +1,362 @@
+	"strconv"
+
+	"github.com/alkiranet/alkira-client-go/alkira"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAlkiraF5LoadBalancer() *schema.Resource {
+	return &schema.Resource{
+		Description:   "F5 Load Balancer Service.",
+		CreateContext: resourceF5LoadBalancerCreate,
+		ReadContext:   resourceF5LoadBalancerRead,
+		UpdateContext: resourceF5LoadBalancerUpdate,
+		DeleteContext: resourceF5LoadBalancerDelete,
+		CustomizeDiff: func(ctx context.Context, d *schema.ResourceDiff, m interface{}) error {
+			client := m.(*alkira.AlkiraClient)
+
+			old, _ := d.GetChange("provision_state")
+
+			if client.Provision == true && old == "FAILED" {
+				d.SetNew("provision_state", "SUCCESS")
+			}
+
+			return nil
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Name of the service.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "Description of the service.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"cxp": {
+				Description: "CXP on which the service should be provisioned.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"size": {
+				Description: "Size of the service, one of" +
+					" `SMALL`, `MEDIUM`, `LARGE`" +
+					" `2LARGE`, `5LARGE`.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"segment_ids": {
+				Description: "IDs of segments associated with the service.",
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"billing_tag_ids": {
+				Description: "IDs of billing tags to associate with" +
+					" the service.",
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeInt},
+				Optional: true,
+			},
+			"global_cidr_list_id": {
+				Description: "ID of global CIDR list from which subnets" +
+					" will be allocated for the external network interfaces of" +
+					" instances. These interfaces host the public IP" +
+					" addresses needed for virtual IPs.",
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"prefix_list_id": {
+				Description: "ID of prefix list to use for IP allowlist",
+				Type:        schema.TypeInt,
+				Optional:    true,
+			},
+			"segment_options": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "The segment options as used by your F5 Load Balancer.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"segment_id": {
+							Description: "ID of the segment.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"nat_pool_prefix_length": {
+							Description: "Prefix length of subnets for the segment.",
```

Comment:
Do we have a default value for this? It feels like an value with a default

Reply:
nat pool prefix length has been removed. so I've removed it from TF as well.